src/modules/configurator/components/Topping.tsx

```diff
@@ -1,13 +1,41 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { ToppingsInfo } from "modules/configurator";
+import { useResetRecoilState, useSetRecoilState } from "recoil";
+import { configuratorAtoms } from "./state";
 
 export const Topping: React.FC<ToppingsInfo> = ({ name, price }) => {
+  const [isChecked, setIsChecked] = useState<boolean>(false);
+  const setTopping = useSetRecoilState(configuratorAtoms.toppings);
+  const resetPizzaDiscount = useResetRecoilState(configuratorAtoms.discount);
+
+  useEffect(() => {
+    if (isChecked) {
```

Comment:
We don't use if-else, but you can add just the first if and if the condition is true then return if its not then just trigger the 2nd function, but the you need to trigger the reset function first because if the contition is true if wont trigger.
```
  useEffect(() => {
    resetPizzaDiscount();
    if (isChecked) {
      setTopping((currentState) => [...currentState, { name, price }]);
      return;
    }

    setTopping((currentState) =>
      currentState.filter((topping) => topping.name !== name)
    );
  }, [isChecked]);

```

or you can use the ternary operator:
```
 useEffect(() => {
    const array = isChecked
      ? [...topping, { name, price }] //topping is the atom 
      : topping.filter((topping) => topping.name !== name);

    setToppingArr(array);
  }, [isChecked]);
```